tests/testthat/test-yaml_as_df.R

```diff
@@ -31,4 +31,33 @@ test_that("read prototyped table [PMT-TEST-0248]", {
   expect_error(yaml_as_df(file))
 })
 
+test_that("table contains .row names", {
+  file <- sysfile("table.yml")
+  df <- yaml_as_df(file)
+  expect_is(df, "data.frame")
+  expect_true(names(df)[1] == ".row")
+  df <- yaml_as_df(file, row_var = "ROW")
+  expect_true(names(df)[1] == "ROW")
+  df <- yaml_as_df(file, row_var = NULL)
+  expect_true(names(df)[1] == "study")
+
+})
 
+
+test_that(".row is renamed if it exists", {
+  yaml <- '
+  a:
+    b: 2
+    .row: 1
+    .row_1: 11
+  aa:
+    b: 3
+    .row: 2
+    .row_1: 22
+  '
+  file <- file.path(tempdir(), "yaml-as-df-test-.row-rename")
+  writeLines(text = yaml, con = file)
+  df <- yaml_as_df(file)
```

Comment:
Using a temporary file is of course perfectly fine, but here's an approach with `textConnection()` that avoids touching the disk:

```
diff --git a/tests/testthat/test-yaml_as_df.R b/tests/testthat/test-yaml_as_df.R
index 2007667..4a81f5b 100644
--- a/tests/testthat/test-yaml_as_df.R
+++ b/tests/testthat/test-yaml_as_df.R
@@ -55,9 +55,7 @@ test_that(".row is renamed if it exists", {
     .row: 2
     .row_1: 22
   '
-  file <- file.path(tempdir(), "yaml-as-df-test-.row-rename")
-  writeLines(text = yaml, con = file)
-  df <- yaml_as_df(file)
+  df <- yaml_as_df(textConnection(yaml))
   expect_is(df, "data.frame")
   expect_true(names(df)[1] == ".row_2")
 })
```

Reply:
Thanks; I didn't know about this function. Super helpful.